editor.planx.uk/src/@planx/components/Send/Public.test.tsx

```diff
@@ -1,36 +1,228 @@
 
-it.todo("sets :localAuthority API param correctly based on team or passport");
+beforeEach(() => (act(() => setState({ teamSlug: "testTeam" }))));
+
+afterEach(() => {
+  vi.clearAllMocks();
+  window.history.pushState({}, "", originalLocation);
+  act(() => setState(initialState))
+});
+
+it("displays a warning at /draft URLs", () => {
+  window.history.pushState({}, "", "/draft");
+  const { getByText } = setup(
+    <SendComponent
+      title="Send"
+      destinations={["bops", "uniform"]}
+    />,
+  );
+
+  expect(getByText(/You can only test submissions on/)).toBeVisible();
+});
+
+it("displays a warning at /preview URLs", () => {
+  window.history.pushState({}, "", "/preview");
+  const { getByText } = setup(
+    <SendComponent
+      title="Send"
+      destinations={["bops", "uniform"]}
+    />,
+  );
+
+  expect(getByText(/You can only test submissions on/)).toBeVisible();
+});
+
+it("displays loading messages to the user", async () => {
+  let resolvePromise: (value: any) => void;
+  const promise = new Promise((resolve) => {
+    resolvePromise = resolve;
+  });
+
+  mockAxios.post.mockImplementationOnce(() => promise);
+
+  const { getByText } = setup(
+    <SendComponent
+      title="Send"
+      destinations={["bops", "uniform"]}
+    />,
+  );
+
+  await tick();
+
+  // Initial loading state
+  expect(getByText(/Submitting your application.../)).toBeVisible();
+
+  // Trigger mock API response
+  resolvePromise!({
+    data: hasuraEventsResponseMock,
+    status: 200,
+    statusText: "OK",
+  });
+
+  expect(mockAxios.post).toHaveBeenCalledTimes(1);
+
+  await tick();
+
+  // Final submission state
+  expect(getByText(/Finalising your submission.../)).toBeVisible();
+});
+
+it("calls the /create-send-event endpoint", async () => {
+  setup(
+    <SendComponent
+      title="Send"
+      destinations={["bops", "uniform"]}
+    />,
+  );
+
+  await waitFor(() => expect(mockAxios.post).toHaveBeenCalledTimes(1));
+
+  expect(mockAxios.post).toHaveBeenCalledTimes(1);
+});
+
+it("generates a valid payload for the API", async () => {
+  const destinations: SendIntegration[] = ["bops", "uniform"];
+
+  setup(
+    <SendComponent
+      title="Send"
+      destinations={destinations}
+    />,
+  );
+
+  await waitFor(() => expect(mockAxios.post).toHaveBeenCalledTimes(1));
+
+  const apiPayload = mockAxios.post.mock.calls[0][1];
+
+  destinations.forEach(destination => {
+    expect(apiPayload).toHaveProperty(destination);
+    expect((apiPayload as Record<SendIntegration, unknown>)[destination]).toHaveProperty("localAuthority", "testTeam");
+  });
+});
+
+describe("Uniform overrides for Buckinghamshire", () => {
+  it("converts property.localAuthorityDistrict to the correct format", async () => {
+    act(() => setState({ 
+      teamSlug: "buckinghamshire", 
+      flow,
+      breadcrumbs: {
+        findProperty: {
+          data: {
+            "property.localAuthorityDistrict": ["Some local authority district"]
```

Comment:
nit: more realistic mock here would be `["Buckinghamshire", "Historic district name"]` - eg because I think we rely on being able to _filter council name out of_ `property.localAuthorityDistrict` array !

Reply:
Ah of course - very good catch - best to be as explicit and through as possible here.

Updated in [480d86b](https://github.com/theopensystemslab/planx-new/pull/3843/commits/480d86bc07da13387ab82dde0119f1982e390aba)